### ISSUE TITLE
[DatePicker] DatePicker moves when in certain position and does not update chosen date

### ISSUE THREAD

**Issue by jhpark1014** [mui/mui-x#9895]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [x] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

https://github.com/mui/mui-x/assets/99851347/6b1b3005-a092-4c31-b4ad-2e343664c474

Steps:
1. Open Datepicker leaving just enough space for only the month picker to be opened at the bottom of the browser.
2. Open the year picker and then choose year which will make the datepicker to move on to the month picker.
3. When trying to choose the month, the month picker moves to the bottom and prevents from changing the month unless moving the cursor down to where the month picker moved.

### Current behavior 😯

When trying to choose the month, the month picker moves to the bottom and prevents from changing the month unless I drag and move the cursor down while still clicking the mouse to where the month picker moved.

### Expected behavior 🤔

Instead of the month picker moving, it would be best if it stays where it was

### Context 🔦

I'm just trying to make a calendar date picker so that day, month, year can be chosen

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
Google Chrome 115.0.5790.110
  Output from `npx @mui/envinfo` goes here.
```
</details>
    "react": "^18.2.0",
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@mui/lab": "^5.0.0-alpha.103",
    "@mui/material": "^5.13.6",
    "@mui/x-date-pickers": "^6.10.2",

**Comment by alexfauquette** [comment]
Does this docs solve your issue?

https://mui.com/x/react-date-pickers/custom-components/#popper

**Comment by jhpark1014** [comment]
Works perfectly. Thanks!

**Comment by berk-can** [comment]
> Works perfectly. Thanks!

Hello Im facing same issue but I didnt understand what solved the problem for you can you explaing please

**Comment by alexfauquette** [comment]
It refers to a docs section introduced here: https://github.com/mui/mui-x/pull/9781 But I don't know where it is now @noraleonte do you have an idea?

**Comment by LukasTy** [comment]
@berk-can The problem has been resolved with https://github.com/mui/mui-x/pull/10541.
It has been released in `v6.16.1`. Which package version are you using? 🤔
